src/Platform.Handlers/src/Xamarin.Platform.Handlers/Platform/Android/SwitchExtensions.cs

```diff
@@ -0,0 +1,254 @@
+	{
+		static ColorTrackingColorStateList? mSwitchTrackStateList;
+		// taken from android sourcec ccode
+		static ColorTrackingColorStateList getSwitchTrackColorStateList(this Context context)
+		{
+			if (mSwitchTrackStateList == null)
+			{
+				int[][] states = new int[3][];
+				int[] colors = new int[3];
+				int i = 0;
+				// Disabled state
+				states[i] = new int[] { -AAttribute.StateEnabled };
+				colors[i] = getThemeAttrColor(context, AAttribute.ColorForeground, 0.1f);
+				i++;
+				states[i] = new int[] { AAttribute.StateChecked };
+				colors[i] = getThemeAttrColor(context, AAttribute.ColorControlActivated, 0.3f);
+				i++;
+				// Default enabled state
+				states[i] = new int[0];
+				colors[i] = getThemeAttrColor(context, AAttribute.ColorForeground, 0.3f);
+				i++;
+				mSwitchTrackStateList = new ColorTrackingColorStateList(states, colors);
+			}
+			return mSwitchTrackStateList;
+		}
+
+		static int getThemeAttrColor(Context mContext, int attr)
+		{
+			TypedValue mTypedValue = new TypedValue();
+			if (mContext.Theme?.ResolveAttribute(attr, mTypedValue, true) == true)
+			{
+				if (mTypedValue.Type >= DataType.FirstInt
+						&& mTypedValue.Type <= DataType.LastInt)
+				{
+					return mTypedValue.Data;
+				}
+				else if (mTypedValue.Type == DataType.String)
+				{
+					throw new NotImplementedException();
+				}
+			}
+			return 0;
+		}
+
+		static int getThemeAttrColor(Context mContext, int attr, float alpha)
+		{
+			int color = getThemeAttrColor(mContext, attr);
+			int originalAlpha = AColor.GetAlphaComponent(color);
+			// Return the color, multiplying the original alpha by the disabled value
+			return (color & 0x00ffffff) | ((int)Math.Round(originalAlpha * alpha) << 24);
+		}
+
+		static int[][] _checkedStates = new int[][]
+					{
+						new int[] { -AAttribute.StateEnabled,  },
+						new int[] { AAttribute.StateChecked },
+						new int[0],
+					};
+
+
+		public static void UpdateIsToggled(this ASwitch aSwitch, ISwitch view)
+		{
+			aSwitch.Checked = view.IsToggled;
+		}
+
+		//static Drawable? _defaultTrackDrawable;
+		public static void UpdateOnColor(this ASwitch aSwitch, ISwitch view)
+		{
+			var onColor = view.OnColor;
+
+			if (aSwitch.Context == null)
+				return;
+
+			if (aSwitch.TrackTintList == null)
+				aSwitch.TrackTintList = getSwitchTrackColorStateList(aSwitch.Context);
+
+			if(aSwitch.Checked && aSwitch.TrackTintList is ColorTrackingColorStateList csl)
+			{
+				int nativeOnColor = 0;
+				if(onColor.IsDefault)
+				{
+					nativeOnColor = getThemeAttrColor(aSwitch.Context, AAttribute.ColorControlActivated, 0.3f);
+				}
+				else
+				{
+					nativeOnColor = onColor.ToNative();
+				}
+
+				var newList = csl.CreateForState(
+					new int[] { AAttribute.StateChecked },
+					nativeOnColor);
+
+				if (newList != aSwitch.TrackTintList)
+					aSwitch.TrackTintList = newList;
+			}
+		}
+
+		public static void UpdateOnColorSimple(this ASwitch aSwitch, ISwitch view)
```

Comment:
@mattleibow @jsuarezruiz so this is the really simple one that works and just toggles based on state. So this is called every single time the checked gets changeed

Reply:
How does this work when the switch is halfway? Is it different to using the state lists?

Reply:
Not sure, I haven't stared deeply into the colors yet :-)  In a perfect world we could project the VSM over the native platform. 

Though, I think with either implementation we'll need some aspect of what I've done in this PR with extracting the colors out via the attributes and copying the android source.  I'm not really sure how else we can rewind the drawable tints to default. The only other approach I found to work was to make a deep copy of the drawable but that seems like a bad idea from a performance stand point